Feature/15 write player class

dustinecker opened the pull request:
We may need to go back and review the buildHotel and buildHouse methods after the BoardSquare classes have all been created to make sure they all work together.

kernohad commented on Model/src/GamePackage/Player.java:
```diff
@@ -1,4 +1,265 @@
+			amountPaid = amount - wallet;
+			wallet -= amountPaid;
+		}
+		
```
Look over the logic for the else. I think it should be amountPaid = wallet and then wallet -= amountPaid, which can be simply wallet = 0. Right? If amount desired is $30, but player only has $10, we dont want amountPaid to be $20, we want to pay all the player has which is $10 and set wallet to 0.

dustinecker replied:
Good catch you are right! Sorry about that.

kernohad commented on Model/src/GamePackage/Player.java:
```diff
@@ -1,4 +1,265 @@
+	public void sell(BoardSquare property) {
+		propertiesOwned.remove(property);
+	}
+	
```
I think we can remove the sell() function. I am having a hard time remembering why we had this here. The original issue has:
- giveProperty(): which should removes a property. The idea here is we call Player.giveProperty() so its removed 
- recieveProperty(): which should add a property. The idea here is we call Player.receiveProperty() so its added. 

I think it makes more sense for the Game class to have a sell method that makes calls to this. We should all have a discussion about this though.

quirogas replied:
the sell method was the way the bank collected payments from players.

kernohad replied:
I forget, why cant we just use pay()? Cant Game say x player needs to pay bank, call x.pay(amount) and give amount to bank?

quirogas replied:
remove sell method

kernohad commented on Model/src/GamePackage/Player.java:
```diff
@@ -1,4 +1,265 @@
+		wallet += property.getMortgageValue();
+		property.setIsMortgaged(true);
+	}
+	
```
mortgage() seems good, but I am thinking if we move sell() to Game, should we move this as well? Again, we need to have a group discussion about this. Probably not in the comments. (probably via slack or in person)

quirogas replied:
Also, as it is, BoardSquare doesn't have a getMortgage() and a setIsMortGaged() methods, so this will not work unless we implement those methods on the BoardSquare class. a Solution would be for the railroad class to extend the PropertySquare class,  and for the parameter to be a PropertySquare.

kernohad commented on Model/src/GamePackage/Player.java:
```diff
@@ -1,4 +1,265 @@
+	public void giveProperty(BoardSquare property) {
+		propertiesOwned.add(property);
+	}
+	
```
The idea was for giveProperty() to be removing a property from the player. As if the player is giving it away. Then there also needs to be a receieveProperty() which adds it.

kernohad commented on Model/src/GamePackage/Player.java:
```diff
@@ -1,4 +1,265 @@
+	 * @param wallet Sets the player's starting wallet value.
+	 *****************************************************************/
+	public Player(String displayName, String token, int wallet) {
+		super();
```
Should it be super(name); and have name as a param as well?

Dcbaas replied:
See my review.

kernohad replied:
Yeah, disregard this comment! ;)

Dcbaas commented on Model/src/GamePackage/Player.java:
```diff
@@ -1,4 +1,265 @@
+	 * @param wallet Sets the player's starting wallet value.
+	 *****************************************************************/
+	public Player(String displayName, String token, int wallet) {
+		super();
```
Remove super(). Its auto generated by eclipse and shouldn't be there

kernohad replied:
Doesn't it need to be there if we aren't using eclipse? But shouldn't it be super(name); and name be a param?

Dcbaas replied:
When auto generating a constructor using eclipse, it generates that super() line a when the constructor is built. Player isn't the subclass of anything and therefore the super() call isn't necessary.  super(name) is in my PropertySquare Class because I called the Base class constructor of PropertySquare which has only one parameter which is the name of the board square.

kernohad replied:
Oh duh, LOL. Sorry I was also working on a subclass of BoardSquare and got confused.

Dcbaas commented on Model/src/GamePackage/Player.java:
```diff
@@ -1,4 +1,265 @@
+ * and other stats.
+ * 
+ * @author Dustin Ecker
+ * @version 2/11/2018
```
Update the date to be today?

Dcbaas commented on Model/src/GamePackage/Player.java:
```diff
@@ -1,4 +1,265 @@
 public class Player {
+	
+	/** Player's name */
+	private String displayName;
```
Should the display name be final?

kernohad replied:
I agree

quirogas replied:
I though we where going to allow the player to choose their own name ?

kernohad replied:
We are, once its chosen and set, it wont change. right?

Dcbaas commented on Model/src/GamePackage/Player.java:
```diff
@@ -1,4 +1,265 @@
+	private String displayName;
+	
+	/** Player's token */
+	private String token;
```
Should the token the player uses be final?

kernohad replied:
I agree

quirogas replied:
same thing again, I though we where having different tokens.

kernohad replied:
Once the token is chosen it wont change. Right?

dustinecker commented:
Fixes:
Changed PLAYER_NAME and TOKEN to final variables
Removed setters for these variables
Fixed issue with pay method
Fixed give method
Added receive method
Removed super from constructor method
Updated to today's date

Still in question:
What should we do with pay method? It's the exact same method as receive now.

quirogas commented on Model/src/GamePackage/Player.java:
```diff
@@ -1,4 +1,257 @@
+	 * @param property The property that the hotel will be built on.
+	 *****************************************************************/
+	public void buildHotel(PropertySquare property) {
+		property.setNumHotels(1);
```
this should be property.setHasHotel(true);

dustinecker commented:
Fixes:
Fixed buildHotel method.
Added sellHouse method.
Added sellHotel method.

Still in question:
I don't think that mortgage should remove the property from player's list of properties because you still own it. It just becomes inactive.

quirogas commented on Model/src/GamePackage/Player.java:
```diff
@@ -1,4 +1,280 @@
+		canMove = true;
+		isBankrupt = false;
+		
+		propertiesOwned = new ArrayList<BoardSquare>();
```
ArrayList of PropertySquare

dustinecker commented:
Removed sell method and updated BoardSquare to PropertySquare for array lists.
